test/plcr.js

```diff
@@ -0,0 +1,603 @@
+const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+const keccak256 = require('js-sha3').keccak_256
+
+const getContract = name => artifacts.require(name)
+const getEvent = (receipt, event, arg) => { return receipt.logs.filter(l => l.event == event)[0].args[arg] }
+const pct16 = x => new web3.BigNumber(x).times(new web3.BigNumber(10).toPower(16))
+
+contract('PLCR', ([owner, voter1, voter2, _]) => {
+  let app, staking
+
+  const voteQuorum = pct16(50)
+  const minorityBlocSlash = pct16(80)
+  const commitDuration = 1000
+  const revealDuration = 1000
+  const stake = 100
+  const lockId = 1
+  const salt = 'salt'.repeat(8)
+
+  const TIME_UNIT_BLOCKS = 0
+  const TIME_UNIT_SECONDS = 1
+
+  const secretHash = (voteOption) => {
+    let node = keccak256(voteOption ? '1' : '0')
+    node = keccak256(new Buffer(node + keccak256(salt), 'hex'))
+    return '0x' + node
+  }
+
+  context('Regular app', async() => {
+    beforeEach(async () =>{
+      staking = await getContract('StakingMock').new()
+
+      app = await getContract('PLCRMock').new()
+      await app.initialize(staking.address, voteQuorum, minorityBlocSlash, commitDuration, revealDuration)
+    })
+
+    it('checks initial values are right', async () => {
+      assert.equal(await app.staking.call(), staking.address, "Staking address should match")
+      assert.equal((await app.voteQuorum.call()).toString(), voteQuorum.toString(), "voteQuorum should match")
+      assert.equal((await app.minorityBlocSlash.call()).toString(), minorityBlocSlash, "minorityBlocSlash should match")
+      assert.equal((await app.commitDuration.call()).toString(), commitDuration, "commitDuration should match")
+      assert.equal((await app.revealDuration.call()).toString(), revealDuration, "revealDuration should match")
+    })
+
+    it('fails on reinitialization', async () => {
+      return assertRevert(async () => {
+        await app.initialize(staking.address, voteQuorum, minorityBlocSlash, commitDuration, revealDuration)
+      })
+    })
+
+    // aux action functions
+
+    const createVote = async () => {
+      const r = await app.newVote("", "", { from: voter1 })
+      const voteId = getEvent(r, "NewVote", "voteId")
+
+      return voteId
+    }
+
+    const commitVote = async (voter, voteOption) => {
+      const voteId = await createVote()
+      // mock lock
+      const startLock = await app.getTimestampExt.call()
+      const endLock = startLock.add(commitDuration + revealDuration + 1)
+      await staking.setLock(stake, TIME_UNIT_SECONDS, startLock, endLock, app.address, "")
+
+      const receipt = await app.commitVote(voteId, secretHash(voteOption), lockId, { from: voter })
+
+      return { voteId: voteId, receipt: receipt }
+    }
+
+    const revealVote = async (voter, voteOption) => {
```

Comment:
Should be named `commitAndReveal` vote for clarity of what the function is really doing.